src/Http/ServerRequest.php

```diff
@@ -1410,12 +1422,26 @@ public function query($name = null)
     /**
      * Read a specific query value or dotted path.
      *
-     * @param string $name The name or dotted path to the query param.
-     * @param mixed $default The default value if the named parameter is not set.
+     * Developers are encouraged to use getQueryParams() when possible as it is PSR-7 compliant, and this method
+     * is not. You can future proof your code by using the following example.
```

Comment:
@dereuromark If you want to use Guzzle as your middleware then getQuery is not a PSR-7 method, and won't exist in the request object. The point of the comment is to advise developers that if they want to be PSR-7 compliant, then don't use getQuery. It's just that ServerRequestInterface is the lower common interface between Guzzle and Cake.

I'm not saying the method will go away. But if you write your code as in the example, then you're code is more PSR-7 compliant. 

I think this was the whole point of PSR-7 support in 3.4, but I'm not completely sure on how it's going to work. I don't think we're completely there yet (i.e. you can't use Guzzle as the middleware).

Maybe I'm wrong about all this. I don't know.